k2/csrc/fsa_renderer.cc

```diff
@@ -68,7 +72,7 @@ std::string FsaRenderer::Render() const {
   os << GeneratePrologue();
 
   for (int32_t i = 0; i != num_states - 1; ++i) {
-    os << ProcessState(fsa_, i);
+    os << ProcessState(fsa_, i, arc_weights_);
   }
 
   // now for the final state
```

Comment:
sometimes the weight for the final state is not zero, so it should also handle this case.

Reply:
@csukuangfj  can you be more explicit about what you mean here?  Or perhaps you misunderstand..
The final state doesn't have a weight, there is no way to express it.  Arcs to the final state have weights on them, which can be interpreted as final weights.

Reply:
.. in fact, if we were to convert to OpenFST models we would change the format; we'd remove the -1 labels and instead treat those arcs as final-probs; and we would not have the final-state.

Reply:
you are right.